kythe/go/platform/analysis/local/local.go

```diff
@@ -140,3 +145,103 @@ type kzipFetcher struct{ r *kzip.Reader }
 
 // Fetch implements the required method of analysis.Fetcher.
 func (k kzipFetcher) Fetch(_, digest string) ([]byte, error) { return k.r.ReadAll(digest) }
+
+// localAnalyzer provides an implementation of the analysis.CompilationAnalyzer
+// interface that works by saving the compilation unit to a temporary kzip file
+// and invoking an indexer binary on it.
+type localAnalyzer struct {
+	cmd     []string
+	fetcher analysis.Fetcher
+}
+
+// NewLocalAnalyzer creates an analyzer that will execute the given command when
+// Analyze() is called. The filepath to a kzip is appended to the command, so if
+// given ["my_indexer", "--index"], the final executed command will be
+// ["my_indexer", "--index", "some/file.kzip"]. The indexer command should emit
+// length-delimited Entry protos to stdout as described in
+// //kythe/go/platform/delimited.
+func NewLocalAnalyzer(cmd []string, fetcher analysis.Fetcher) (*localAnalyzer, error) {
+	if len(cmd) == 0 {
+		return nil, fmt.Errorf("parameter 'cmd' must not be empty")
+	}
+	return &localAnalyzer{cmd: cmd, fetcher: fetcher}, nil
+}
+
+// Analyze implements the analysis.CompilationAnalyzer interface.
+func (a *localAnalyzer) Analyze(ctx context.Context, req *apb.AnalysisRequest, f analysis.OutputFunc) error {
+	tmpkzip, err := saveSingleUnitKzip(ctx, req.Compilation, a.fetcher)
+	if err != nil {
+		return fmt.Errorf("creating tmp kzip: %v", err)
+	}
+	defer os.Remove(tmpkzip)
+
+	// Run indexer binary
+	var errBuf strings.Builder
+	var outBuf bytes.Buffer
+	cmd := exec.Command(a.cmd[0], append(a.cmd[1:], tmpkzip)...)
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &errBuf
+	if err = cmd.Run(); err != nil {
+		return fmt.Errorf("indexing kzip %s: %v. stderr=%s", tmpkzip, err, errBuf.String())
+	}
+
+	// Deserialize output and pass it to `f`.
+	rd := delimited.NewReader(&outBuf)
+	for {
+		n, err := rd.Next()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return fmt.Errorf("parsing entry stream: %v", err)
+		}
+
+		if err := f(ctx, &apb.AnalysisOutput{Value: n}); err != nil {
+			return fmt.Errorf("unable to write analysis output: %v", err)
+		}
+	}
+	if err := f(ctx, &apb.AnalysisOutput{FinalResult: &apb.AnalysisResult{Status: apb.AnalysisResult_COMPLETE}}); err != nil {
+		return fmt.Errorf("unable to write analysis output: %v", err)
+	}
+
+	return nil
+}
+
+// saveSingleUnitKzip creates a kzip file containing the given compilation unit
+// and any files it depends on. File contents are retrieved from the given
+// fetcher object.
+func saveSingleUnitKzip(ctx context.Context, cu *apb.CompilationUnit, fetcher analysis.Fetcher) (path string, err error) {
```

Comment:
```suggestion
func saveSingleUnitKzip(ctx context.Context, cu *apb.CompilationUnit, fetcher analysis.Fetcher) (_ string, err error) {
```

Since you don't actually need `path` for anything in the function, but you do want `err` for the deferral.